test/functional/mining_testnet.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mining on testnet
+
+Test mining related RPCs that involve difficulty adjustment, which
+regtest doesn't have.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+from test_framework.blocktools import (
+    DIFF_1_N_BITS,
+    DIFF_1_TARGET,
+    DIFF_4_N_BITS,
+    DIFF_4_TARGET,
+    nbits_str,
+    target_str
+)
+
+import os
+
+class MiningTestnetTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.chain = "testnet4"
+
+    def add_options(self, parser):
+        parser.add_argument(
+            '--datafile',
+            default='data/testnet4.hex',
```

Comment:
In commit "test: check testnet4 difficulty adjustment" (6b504d6cf8d0b3cf14b0df1683d080f505588e6b)

Can there be some comment about how the testnet.4 file is generated and what it contains? Maybe there could be a readme in the test/functional/data/ directory.

It seems ok to have one test like this with a large static data file, but it would be nice in the future to be able to generate block data more dynamically, and having a description of the data could help implement that.

Reply:
Just this embarrassing bash script:

```sh
for i in {0..2015}
do                                 
hash=`build/src/bitcoin-cli -testnet4 getblockhash $i`
block=`build/src/bitcoin-cli -testnet4 getblock $hash 0`
echo $block >> test/functional/data/testnet4.hex
done
```